### DISCUSSION TITLE
How to use insuffisant solar production ?

### DISCUSSION THREAD

**Discussion by Superdada44** [jmcollin78/solar_optimizer#81] [Ideas]
Hello,

I'm so happy to use Solar Optimizer and I would like to contribute to his enhencement so I would share here my reflexion. I'm not sure that my idea is a better than that the one is used now, so feel free to comment. With the winter time, I have a lot of days when I have solar production but just not enough to start my water heater (like 1800W when 2000W needed).
Another way to calculate the best objectif would be to compare when it's better to sell all my solar production and use offpeak hours (and price), than to switch on the device even solar production cover only a part of what is needed.

Condition could be:
(power_max * offpeak_buy_price) – (exported_power * sell_price) >= (imported_power  * Peak_buy_price)
cost_to_run_at_offpeak - income_sell_production >= cost_to_switch_on_now

My example:
Water heater power_max : 2kW
Solar Production: 1.8W
offpeak_buy_price: 0.22
Peak_buy_price: 0.32
sell_price: 0.176

(2kW * 0.22) - (1.8W * 0.176) >= ((2kW - 1.8kW) * 0.32)
In this case 0.1232 >= 0.064 so it's cheaper to switch on now even solar production don't cover all it's needed

Thanks a lot !

**Comment by jmcollin78** [comment]
Hello @Superdada44,

This is interesting. At first though SO was to optimize the solar production consumption and not to optimize your electricity bill.
But I will have a look more closer when I have time. I think there is drawnack in your method for people that just want to optimize their solar consumption (and don't care about offpeak)

**Comment by Superdada44** [comment]
Hello Jean-Marc,

Thanks for reading my post. Please only consider the cost as a way to have an objective view when it's better to switch on even the production dont cover all what you need.

Best regards

David

**Comment by sleveille4** [comment]
hello, i concur with above request as i thought this is actually what this add-on was for.

this would really be nice as this is 100% my use case, small solar array (1kw peak) and no battery, no selling.

i try to make an automation to achieve use solar + day priced elec when it is cheaper than night price only.
AND to set heating time target, so that of it only partially heat with solar, it will complete the remaining heating process at night cost

but while it seems to work (so far at least until something doesn't :) ) it's messy automation (my fault because 3 modes: guest / summer / winter; and EDF TEMPO conditions) + several helpers (filters / on time)
